jkube-kit/generator/java-exec/src/test/java/org/eclipse/jkube/generator/javaexec/JavaExecGeneratorTest.java

```diff
@@ -50,6 +54,136 @@ void setUp() {
     when(generatorContext.getProject().getPlugins()).thenReturn(plugins);
   }
 
+
+  @Test
+  @DisplayName("Warn if final output artifact is missing")
+  void warnIfMissingFinalOutputArtifact() throws IOException {
+    // @Todo: move these initializations to setup method when replacing mocks with real objects
+    KitLogger testLogger = spy(new KitLogger.SilentLogger());
+    targetDir = Files.createDirectory(temporaryFolder.resolve("target")).toFile();
+    project = JavaProject.builder()
+            .baseDirectory(targetDir)
+            .buildDirectory(targetDir.getAbsoluteFile())
+            .buildPackageDirectory(targetDir.getAbsoluteFile())
+            .outputDirectory(targetDir)
+            .buildFinalName("sample")
+            .packaging("jar")
+            .build();
+    generatorContext = GeneratorContext.builder()
+            .logger(new KitLogger.SilentLogger())
+            .project(project)
+            .logger(testLogger)
+            .build();
+
+    // Given, When
+    new JavaExecGenerator(generatorContext).checkMissingArtifacts();
```

Comment:
I don't see the difference with the BaseGeneratorTest here. It is doing the same thing and calling the same method. I would rather make sure that the method is called and well executed when calling createAssembly()